docs/aws/audit/sgaudit/rules/mq_active_deployment_mode_remediation.mdx

```diff
@@ -1,6 +1,239 @@
+
+3. **Restrict Access:**
+   - Edit the inbound rules to restrict access to only trusted IP addresses or specific VPCs.
+   - For example, if your MQ service should only be accessed from a specific IP range, add a rule that allows access only from that range and remove any broader rules.
+
+4. **Enable Logging and Monitoring:**
+   - Enable VPC Flow Logs to capture information about the IP traffic going to and from network interfaces in your VPC.
+   - Use AWS CloudTrail to monitor API calls and detect any unauthorized changes to your Security Groups.
+
+By following these steps, you can ensure that your Security Groups are configured correctly to prevent unauthorized access to your MQ service.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration of "MQ Active Has Deployment Mode" in Security Groups using AWS CLI, you can follow these steps:
+
+1. **Create a Security Group with Specific Rules:**
+   Ensure that you create a security group with the least privilege principle, allowing only necessary traffic.
+
+   ```sh
+   aws ec2 create-security-group --group-name MySecureGroup --description "My secure group for MQ"
+   ```
+
+2. **Add Inbound Rules to Restrict Access:**
+   Add inbound rules to the security group to restrict access to only trusted IP addresses and necessary ports.
+
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-name MySecureGroup --protocol tcp --port 5671 --cidr <trusted_ip_range>
+   ```
+
+3. **Add Outbound Rules to Restrict Traffic:**
+   Add outbound rules to the security group to restrict traffic to only necessary destinations.
+
+   ```sh
+   aws ec2 authorize-security-group-egress --group-name MySecureGroup --protocol tcp --port 5671 --cidr <trusted_ip_range>
+   ```
+
+4. **Review and Update Security Group Rules Regularly:**
+   Regularly review and update the security group rules to ensure they comply with the least privilege principle and remove any unnecessary rules.
+
+   ```sh
+   aws ec2 describe-security-groups --group-names MySecureGroup
+   ```
+
+By following these steps, you can prevent misconfigurations in your security groups related to MQ Active Deployment Mode using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of "MQ Active Has Deployment Mode" in Security Groups using Python scripts, you can follow these steps:
+
+### 1. **Set Up AWS SDK (Boto3)**
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. **Initialize Boto3 Client**
+Initialize the Boto3 client for EC2, which will allow you to interact with Security Groups.
+
+```python
+import boto3
+
+# Initialize a session using Amazon EC2
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>